observability/WFLY-18582_add-prometheus-endpoint-to-micrometer-extension.adoc

```diff
@@ -0,0 +1,123 @@
+---
+categories:
+- observability
+stability-level: community
+issue: https://github.com/wildfly/wildfly-proposals/issues/590
+feature-team:
+ developer: jasondlee
+ sme:
+  - jamezp
+ outside-perspective:
+  - kabir
+---
+= [Community] WFLY-18582 - Add prometheus endpoint to micrometer extension
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Micrometer replaced Micrometer Metrics as the user-facing metrics library provided by WildFly. As part of that
+transition, the export behavior moved from a pull-based Prometheus-like format to push-based mechanism using the
+OpenTelemetry Protocol. Based on feedback from users who still desire to use the pull-based approach, this change will
+add support for the Prometheus Registry in addition to or in place of the OTLP Registry.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-18582[WFLY-18582]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-2234[EAP7-2234]
+* https://issues.redhat.com/browse/EAP7-2235[EAP7-2235]
+
+
+=== Affected Projects or Components
+
+Only the Micrometer subsystem and related tests will be affected
+
+=== Other Interested Projects
+
+- OpenShift has an interest in Prometheus metrics
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* OpenShift Source-to-Image (S2I)
+* Managed domain
+* Bootable jar
+
+== Requirements
+
+* Existing servers, if configured to use Micrometer (e.g., the `endpoint` property has been set), will be migrated to be
+  an OTLP-based configuration. Since OTLP is the only current option, there is no ambiguity in this migration.
+* Existing servers with the extension enabled but with no endpoint set will not be changed. That is to say, Micrometer
+  will continue to use a no-op registry.
+* Support for a new registry, `PrometheusRegistry`, will be added.
+** It can be added as the only registry, or it can be added _in addition to_ the OTLP Registry
+** The registry will expose two model attribute:
+*** `context`: the context (e.g., `/prometheus`) on which the metrics will be exposed. This is undefined by default and
+  must be specifed by the administrator when adding the resource.
+*** `security-enabled`: whether to secure the context using the user and roles defined in the application server. The context
+  will be secured by default
+** If the user would like to use `/metrics`, the WildFly metrics extension must be disabled first to avoid a naming conflict.
```

Comment:
Is this going to be traced in logs, if the WildFly metrics are not disabled?

Reply:
Yes, there will be logging and the server will fail to start due to the conflict. For the record, a cleaner, earlier-in-the-boot-process handling has been merged upstream, but that is not part of this backport.